### PR TITLE
Enhancement: Deprecating use of langspec 

### PR DESCRIPTION
**Pull request by ahangsu**

### CODE REVIEW

File: tests/steps/steps.py
Comment (ahangsu) [inline tests/steps/steps.py:1019]:
```suggestion
```

File: .test-env
Comment (ahangsu) [inline .test-env:4]: after algorand-sdk-testing PR merged, do
```suggestion
SDK_TESTING_BRANCH="master"
```

File: algosdk/logic.py
Comment (michaeldiamant) [inline algosdk/logic.py:60]: @ahangsu Not a fully formed thought, but I've run out of time for this review cycle:  When reviewing Python-specific deprecation methods did you assess https://docs.python.org/3/library/exceptions.html#DeprecationWarning?

Looking for your assessment based on what I've found.

* From my partial read, the warning _is_ the Python-way of adding a deprecation warning.  Whether or not the docstring should _also_ be updated is unclear to me.
* I'm also noting there is a prior example in the repo using the warning: https://github.com/algorand/py-algorand-sdk/blob/9f496246124090157770da4b6dc012f5de0e1275/algosdk/mnemonic.py#L93-L97
* I began reading https://docs.python.org/3/library/warnings.html, but haven't finished.  It explains how/where the warning appears.  
  * For example, the example warning I shared above appears in the build:  https://app.circleci.com/pipelines/github/algorand/py-algorand-sdk/890/workflows/a84d0e9a-76ce-470a-8109-e32112fce5a1/jobs/4012?invite=true#step-104-22.

Comment (ahangsu) [reply algosdk/logic.py:60]: makes sense, I look into the `DeprecationWarning` and will add a similar line in `mnemonic.py` in these deprecated methods.

Comment (ahangsu) [reply algosdk/logic.py]: now the `DeprecationWarning` is reflected in CI build log: https://app.circleci.com/pipelines/github/algorand/py-algorand-sdk/894/workflows/5422bf31-4108-4c90-8260-e71b32bfd3b4/jobs/4042

Comment (michaeldiamant) [reply algosdk/logic.py]: @ahangsu Thanks for the change - resolving.

For completeness - It's unclear to me if we still also need (or want) to mention deprecation in the docstring.  Since it's already been done, I'm opting to suggest no further changes.

**Review by jasonpaulos — COMMENTED** [review]
Looks good, I will hold off on approving until sdk-testing branch is updated

**Review by michaeldiamant — APPROVED** [review]

**Review by jasonpaulos — APPROVED** [review]

**Pull request merged**
